packages/lib-ts/src/log.ts

```diff
@@ -0,0 +1,83 @@
+// eslint-disable-next-line no-secrets/no-secrets
+// This file is based on code from "The Graph Tooling" (https://github.com/graphprotocol/graph-tooling/tree/7faa3098b2e6c61f09fc81b8b2d333e66b0080d1).
+// Licensed under the MIT License.
+// Copyright (c) 2018 Graph Protocol, Inc. and contributors.
+// Modified by Mimic Protocol, 2025.
+
+export namespace log {
+  @external('log', '_log')
+  declare function _log(level: Level, msg: string): void
+
+  export enum Level {
+    CRITICAL = 0,
+    ERROR = 1,
+    WARNING = 2,
+    INFO = 3,
+    DEBUG = 4,
+  }
+
+  /**
+   * Logs a critical message that terminates the execution.
+   *
+   * @param msg Format string like "Value = {}, other = {}".
+   * @param args Format string arguments.
+   */
+  export function critical(msg: string, args: Array<string>): void {
```

Comment:
We can add a default for args in all the logs so you don't have to send an empty string if you don't need args

```suggestion
  export function critical(msg: string, args: Array<string> = []): void {
```